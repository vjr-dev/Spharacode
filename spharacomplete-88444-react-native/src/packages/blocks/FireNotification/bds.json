{
  "bds": {
    "name": "FireNotification",
    "id": "FireNotification",
    "platform": [
      "desktop",
      "mobile"
    ],
    "author": "CMT",
    "summary": null,
    "license": "BSD",
    "version": "1.0.0",
    "pages": [
      {
        "skip-to-login": {
          "hotspots": [
            
            {
              "type": "navigation",
              "id": "btnNavigate",
              "options": [
                "target",
                "navigationTransition",
                "navigationType",
                "enabled"
              ]
            }
          ]
        }
      }
    ]
  }
}
