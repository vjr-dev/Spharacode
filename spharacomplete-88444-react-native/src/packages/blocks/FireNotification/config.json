{
  "id": "FireNotification",
  "verion": "0.0.1",
  "status": "preAlpha",
  "path": "./packages/blocks/FireNotification",
  "assembler-adapters": [
    "FireNotificationAdapter"
  ],
  "mobile": {
    "merge-engine-navigation": {},
    "assembler-exports": [
      "FireNotification"
    ],
    "assembler-dependencies": [],
    "catalog-dependencies": [
      "core",
      "FireNotification"
    ],
    "assembler-devDependencies": [],
    "merge-engine-screens": {
      "FireNotification": {
        "file": "FireNotification.tsx",
        "extends": "FireNotificationController",
        "configURL": "",
        "validationRulesURL": "",
        "imports": [
          {
            "value": "FireNotificationController",
            "path": "./FireNotificationController"
          },
          {
            "value": "{Props}",
            "path": "./FireNotificationController"
          }
        ]
      }
    }
  }
  
}
