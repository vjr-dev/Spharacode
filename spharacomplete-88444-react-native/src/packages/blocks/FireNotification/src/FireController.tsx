// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { IBlock } from "../../../framework/src/IBlock";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
    getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";


import {  Linking, Platform } from "react-native";
import AsyncStorage from '@react-native-async-storage/async-storage'
import RNImmediatePhoneCall from 'react-native-immediate-phone-call';
import { GotoHomePage } from "../../../components/src/Navigation/NavigationFunctions";



export const configJSON = require("./config");


export interface Props {
    navigation: any;
    id: string;
    
    
}

interface S {
    
    contacttdata: any
    Address: any
    Time: any
    
}

interface SS {
    id: any;
    
    
}

export default class MakeDonationController extends BlockComponent<Props, S, SS> {
    
    FiredataID: any = "";
    

    constructor(props: Props) {
        super(props);
        this.receive = this.receive.bind(this);

        
        this.subScribedMessages = [
            
            getName(MessageEnum.RestAPIResponceMessage),
            getName(MessageEnum.RestAPIRequestMessage),
            getName(MessageEnum.RestAPIResponceEndPointMessage),
            getName(MessageEnum.RestAPIRequestBodyMessage),
            getName(MessageEnum.RestAPIRequestHeaderMessage),
            getName(MessageEnum.RestAPIRequestMethodMessage),
            
        ];

        this.state = {
            
            contacttdata: [],
            Address: "",
            Time: "",

            
        };
        runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);

        
        
    }

    

    async componentDidMount() {
        this.setState({ contacttdata: this.props.route.params.DDA, Address: this.props.route.params.Address, Time: this.props.route.params.Time })
    }
   
    async onFireCall() {
        const data = await AsyncStorage.getItem("User_Data")
        if (data != null) 
        {
            if (Platform.OS == "android") {
                RNImmediatePhoneCall.immediatePhoneCall(JSON.parse(data).data.attributes.fire)
            } else {
                Linking.openURL(`telprompt:${JSON.parse(data).data.attributes.fire}`)

            }
        }
    }

    doneclick() {
        GotoHomePage(this.props);
    }
    goback() {
        this.props.navigation.goBack()
    }

    
}
// Customizable Area End
