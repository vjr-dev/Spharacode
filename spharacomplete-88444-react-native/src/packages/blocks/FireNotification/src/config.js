// Customizable Area Start
Object.defineProperty(exports, "__esModule", {
    value: true,
});
exports.FireApiContentType = "application/json";
exports.FireAPiEndPoint = "bx_block_dashboard/dashboards/fire_incident";
exports.FireAPiMethod = "POST";
// Customizable Area End