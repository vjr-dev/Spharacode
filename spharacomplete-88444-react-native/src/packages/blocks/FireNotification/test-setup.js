// test-setup.js
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NativeModules } from 'react-native';

configure({ adapter: new Adapter() });



NativeModules.RNCAsyncStorage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    mergeItem: jest.fn(),
    clear: jest.fn(),
    getAllKeys: jest.fn(),
    flushGetRequests: jest.fn(),
    multiGet: jest.fn(),
    multiSet: jest.fn(),
    multiRemove: jest.fn(),
    multiMerge: jest.fn(),
};

jest.mock('react-native/Libraries/Linking/Linking', () => ({
    openURL: jest.fn(),
  }));

  jest.mock('react-native/Libraries/Utilities/Platform', () => ({
    OS: 'macos',
    select: () => null
}));

jest.mock('react-native-immediate-phone-call', ()=> {
    return {
        immediatePhoneCall : jest.fn().mockImplementation(() => { return {} })
    }
});