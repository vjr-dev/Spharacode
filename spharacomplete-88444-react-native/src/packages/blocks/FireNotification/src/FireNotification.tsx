// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";

import {
    StyleSheet,
    Text,
    View,
    ScrollView,
    
    Image,
    ImageBackground,
    Dimensions,
    SafeAreaView,
    TouchableOpacity,
    
} from "react-native";

import MakeDonationController, { Props } from "./FireController";
import { back1, back2, image_call, image_fire } from "./assets";
import { RFValue } from "react-native-responsive-fontsize";
const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;
import { COLORS } from "../../../framework/src/Globals";




export default class Donation extends MakeDonationController {
    constructor(props: Props) {
        super(props);
    }

    
    

    render() {
        let Firetime = this.state.Time
        return (

            <SafeAreaView style={styles.container1}>
                <View style={styles.child}>
                    <ImageBackground
                        source={back1}
                        style={styles.image}>
                        <ImageBackground
                            source={back2}
                            style={styles.image2}>
                            <ScrollView showsVerticalScrollIndicator={false}>
                                <View style={{ width: windowWidth, backgroundColor: COLORS.darkorange, alignItems: "center", justifyContent: "center" }}>

                                    <Image
                                        source={image_fire}
                                        style={{ height: RFValue(75), marginTop: 50, width: RFValue(75), tintColor: COLORS.white }}
                                    />
                                    <Text style={{ paddingTop: 12, color: COLORS.white, fontWeight: "700", fontSize: 20 }}>Fire dispatch is confirmed.</Text>
                                    <Text style={{ color: COLORS.white, fontWeight: "700", fontSize: 20 }}>Help is on the way.</Text>

                                    <Text style={{ paddingTop: 15, color: COLORS.white, fontWeight: "400", textAlign: "center" }}>Sent to {this.state.Address}.</Text>
                                    <Text style={{ color: COLORS.white, fontWeight: "400", textAlign: "center" }}>Sent time: {Firetime}</Text>
                                    <TouchableOpacity testID="onFirecallID" onPress={()=>this.onFireCall()} style={{
                                        height: 50, width: 50, backgroundColor: COLORS.headerbackground, borderRadius: 50, marginTop: 50, alignItems: "center", justifyContent: "center"

                                        , shadowColor: "#000",
                                        shadowOffset: {
                                            width: 0,
                                            height: 10,
                                        },
                                        shadowOpacity: 0.53,
                                        shadowRadius: 13.97,

                                        elevation: 21,

                                    }}>

                                        <Image
                                            source={image_call}
                                            style={{ height: 20, width: 20, tintColor: COLORS.white }}
                                        />
                                    </TouchableOpacity>
                                    <Text style={{ paddingTop: 18, paddingBottom: 25, fontWeight: "700", color: COLORS.headerbackground }}>Call to add info or to cancel fire alert.</Text>

                                </View>
                                <View style={{ width: windowWidth, alignItems: "center", justifyContent: "center" }}>
                                    <Text style={{ paddingTop: 18, color: COLORS.lightwhite, fontWeight: "400", fontSize: 20 }}>We have alerted these contacts.</Text>
                                    <View
                                        style={{ marginTop: 5 }}


                                    >
                                        {this.state.contacttdata.map((item: { item: any, name: any }) => (
                                            <Text style={{ color: COLORS.ultralightwhite, padding: 5 }} key={item.name}>{item.attributes.name}</Text>
                                        ))}
                                    </View>

                                    <TouchableOpacity
                                        onPress={() =>
                                            this.doneclick()
                                        }
                                        testID="doneClickID"
                                        style={{ marginVertical: 25 }}
                                    >
                                        <Text testID="contactName" style={{ color: COLORS.darkorange, fontWeight: "400", fontSize: 18 }}>Done</Text>
                                    </TouchableOpacity>
                                </View>
                            </ScrollView>
                        </ImageBackground>
                    </ImageBackground>
                </View>
            </SafeAreaView>

            
        );
    }
}


const styles = StyleSheet.create({

    container1: { flex: 1, backgroundColor: COLORS.ultradarkorange },
    child: { width: windowWidth, height: windowHeight, flex: 1 },

    image: {
        height: windowHeight, width: windowWidth, flex: 1
    },
    image2: {
        height: windowHeight, width: windowWidth, alignItems: "center", flex: 1
    },

});

//// Customizable Area End