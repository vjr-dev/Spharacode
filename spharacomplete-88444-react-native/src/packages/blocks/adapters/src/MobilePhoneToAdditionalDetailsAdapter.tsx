import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
