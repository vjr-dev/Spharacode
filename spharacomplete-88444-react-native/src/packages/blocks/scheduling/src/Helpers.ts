/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});