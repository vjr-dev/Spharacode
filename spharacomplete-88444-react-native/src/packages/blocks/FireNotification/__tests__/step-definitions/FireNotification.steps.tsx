//@ts-ignore
//@ts-nocheck
import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'
import * as helpers from '../../../../framework/src/Helpers'
import {Linking, Platform,Text} from "react-native";
import RNImmediatePhoneCall from 'react-native-immediate-phone-call';
import AsyncStorage from '@react-native-async-storage/async-storage'
import { COLORS } from "../../../../framework/src/Globals";
import React from "react";
import FireNotification from "../../src/FireNotification"
const navigation = require("react-navigation")

const screenProps = {
    navigation: {
        navigate: jest.fn(),
        goBack: jest.fn(),
        addListener: jest.fn().mockImplementation((event: any, callback: any) => {
            callback();
          }),
        replace:jest.fn(),
    },
    state: {
        params: {
          data: { name: 'Test Name' },
          userName: 'abc',
          contacttdata: [
            { item: 1, name: 'John' },
            { item: 2, name: 'Doe' }
        ],
        DDA: 'your DDA value',
        Address: 'your Address value',
        Time: '8:00 PM',

        }
    },
    

         
  
   
    id: "FireNotification"
  }




const feature = loadFeature('./__tests__/features/FireNotification-scenario.feature');

defineFeature(feature, (test) => {


    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'android' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'android');
    });
    test('User navigates to FireNotification', ({ given, when, then }) => {
        let FireNotiBlockWrapper:ShallowWrapper;
        let instance:FireNotification ; 

        given('I am a User loading FireNotification', () => {
            //@ts-ignore
            FireNotiBlockWrapper = shallow(<FireNotification {...screenProps}/>)
        });

        when('I navigate to the FireNotification', () => {
             instance = FireNotiBlockWrapper.instance() as FireNotification
        });

        then('FireNotification will load with out errors', () => {
            expect(FireNotiBlockWrapper).toBeTruthy()
            expect(FireNotiBlockWrapper).toMatchSnapshot()
        });
        then('I can select the button with with out errors', () => {
            let fireCallButton = FireNotiBlockWrapper.findWhere((node) => node.prop('testID') === 'onFirecallID');
            fireCallButton.simulate('press', instance.onFireCall());
            let doneCallButton = FireNotiBlockWrapper.findWhere((node) => node.prop('testID') === 'doneClickID');
            doneCallButton.simulate('press', instance.doneclick());
            instance.goback()
            const Address = "your Address value"
           const Time = "8:00 PM"
            expect(Address).toEqual('your Address value');
            expect(Time).toEqual('8:00 PM');

           
        
            expect(fireCallButton).toBeTruthy();
        });

        then('renders Text components for each item in contacttdata', () => {
            const contactData = [
                { item: 'item1', name: 'Contact 1' },
                { item: 'item2', name: 'Contact 2' }
              ];
            const textComponents = FireNotiBlockWrapper.find(Text);
            expect(textComponents).toHaveLength(7); // Assuming there are 2 items in contacttdata array
          

        })
        then('should call RNImmediatePhoneCall.immediatePhoneCall for Android', async () => {
            Platform.OS = 'android';
            AsyncStorage.getItem = jest.fn().mockResolvedValue('{"data": {"attributes": {"fire": "123456"}}}');
        
          
            await FireNotiBlockWrapper.instance().onFireCall();
        
            expect(RNImmediatePhoneCall.immediatePhoneCall).toHaveBeenCalledWith('123456');

        })
        then('should call Linking.openURL for iOS', async () => {
            Platform.OS = 'ios';
            AsyncStorage.getItem = jest.fn().mockResolvedValue('{"data": {"attributes": {"fire": "123456"}}}');
        
            
            await FireNotiBlockWrapper.instance().onFireCall();
        
            expect(Linking.openURL).toHaveBeenCalledWith('telprompt:123456');

        })
    })
}
)
