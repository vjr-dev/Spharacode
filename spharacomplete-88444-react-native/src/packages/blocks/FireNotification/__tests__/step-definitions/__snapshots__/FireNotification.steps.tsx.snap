// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FireNotification User navigates to FireNotification 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#fa6738",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/FireNotification/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/FireNotification/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 1334,
            "width": 750,
          }
        }
      >
        <ScrollView
          showsVerticalScrollIndicator={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255,138,73,1)",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/FireNotification/assets/image_fire.png",
                }
              }
              style={
                Object {
                  "height": 147,
                  "marginTop": 50,
                  "tintColor": "rgb(255,255,255)",
                  "width": 147,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "paddingTop": 12,
                }
              }
            >
              Fire dispatch is confirmed.
            </Text>
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 20,
                  "fontWeight": "700",
                }
              }
            >
              Help is on the way.
            </Text>
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontWeight": "400",
                  "paddingTop": 15,
                  "textAlign": "center",
                }
              }
            >
              Sent to 
              .
            </Text>
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontWeight": "400",
                  "textAlign": "center",
                }
              }
            >
              Sent time: 
            </Text>
            <ForwardRef
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#454545",
                  "borderRadius": 50,
                  "elevation": 21,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 50,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 10,
                    "width": 0,
                  },
                  "shadowOpacity": 0.53,
                  "shadowRadius": 13.97,
                  "width": 50,
                }
              }
              testID="onFirecallID"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/FireNotification/assets/image_call.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "tintColor": "rgb(255,255,255)",
                    "width": 20,
                  }
                }
              />
            </ForwardRef>
            <Text
              style={
                Object {
                  "color": "#454545",
                  "fontWeight": "700",
                  "paddingBottom": 25,
                  "paddingTop": 18,
                }
              }
            >
              Call to add info or to cancel fire alert.
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#cdcdcd",
                  "fontSize": 20,
                  "fontWeight": "400",
                  "paddingTop": 18,
                }
              }
            >
              We have alerted these contacts.
            </Text>
            <View
              style={
                Object {
                  "marginTop": 5,
                }
              }
            />
            <ForwardRef
              onPress={[Function]}
              style={
                Object {
                  "marginVertical": 25,
                }
              }
              testID="doneClickID"
            >
              <Text
                style={
                  Object {
                    "color": "rgba(255,138,73,1)",
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
                testID="contactName"
              >
                Done
              </Text>
            </ForwardRef>
          </View>
        </ScrollView>
      </ImageBackground>
    </ImageBackground>
  </View>
</ForwardRef(SafeAreaView)>
`;
